decimal64math_test.go

```diff
@@ -379,3 +379,93 @@ func BenchmarkDecimal64Sub(b *testing.B) {
 		_ = x[i%len(x)].Sub(y[i%len(y)])
 	}
 }
+
+func TestAddOverflow(t *testing.T) {
+	require := require.New(t)
+
+	a := MustParseDecimal64("-9.999999999999999e384") // largest dec64 number
+	b := MustParseDecimal64("0.000000000000001e384")
+	require.Equal(NegInfinity64, a.Sub(b))
+
+	a = MustParseDecimal64("9.999999999999999e384") // largest dec64 number
+	b = MustParseDecimal64("0.000000000000001e384")
+	require.Equal(Infinity64, a.Add(b))
+
+	a = MustParseDecimal64("9.999999999999999e384") // largest dec64 number
+	b = MustParseDecimal64("0.0000000000000001e384")
+	require.Equal(a, b.Add(a))
```

Comment:
I suggest a simpler layout. While I'm loath to write the code for you, this one's much easier to show than describe:
```go
	a = MustParseDecimal64("9.999999999999999e384") // largest dec64 number
	require.Equal(Infinity64, a.Add(MustParseDecimal64("1e369")))
	require.Equal(a, a.Add(MustParseDecimal64("1e369")))
```
You should also test for more borderline edge cases, such as:

- 9.999999999999999e384 + 0.4999999999999999e369
- 9.999999999999999e384 + 0.5e369.

I'm not sure what the standard says here. After scale-matching, do you add then round or round then add, and how many digits do you hold during the calculation? We can use uint128T to make the calculations mathematically precise, but the standard might allow (or require) some precision to be thrown away early to support maximal efficiency and limitations of hardware implementations. Raise an issue to resolve this later.

Either way, the principle is to discern the exact boundary between stays-the-same and jumps-to-∞, and test the two values immediately above and below it.